### ISSUE TITLE
Playback Error (iOS)

### ISSUE THREAD

**Issue by LLinoor** [jellyfin/jellyfin#3957]
(The problem has already been reported on Reddit, but I can't find out if it was reported on GitHub)

When I launch a movie on my iPhone, an error appears with the title "Playback Error" (The movie is a .mkv file)

Link to the Reddit post: https://www.reddit.com/r/jellyfin/comments/i3ipwz/playback_error_on_ios_appweb_browser_since_106/

**System:**
 - OS: Ubuntu 20.04 (Xubuntu)
 - Virtualization: None
 - Client: iOS App
 - Device : iPhone SE (iOS 14 public beta 5)
 - Jellyfin Version: 10.6.3
 - Playback: Direct Play
 - Installed Plugins: None
 - Reverse Proxy: None
 - Base URL: None
 - Networking: Host
 - Storage: Local

**To Reproduce**

1. Play a movie on a iOS Device

**Expected behavior**
Movie must start

**Logs**
Provided in the post Reddit

**Screenshots**
Provided in the post Reddit

**Comment by LLinoor** [comment]
i don't know because the problem is reaching many more people since the last update (given the number of comments on the Reddit post mentioning the same problem)

**Comment by SigHunter** [comment]
just switched over from emby and installed jellyfin 10.6.3 only to notice that all iOS devices can't play anything :(
this should be top priority, please fix asap. major functionality broken

**Comment by Flexusma** [comment]
Same here, running jellyfin on proxmox with the vm being deb 10, jellyfin 10.6.4. Gonna give more info soon.

**Comment by Freeman77** [comment]
Hi, I also have this problem but only with an iPhone 6s, playback works fine with an iPhone 5s using the jellyfin app or web, maybe it's iOS/Safari's fault?

[Server]
OS: Windows 10 Pro, version 2004
Jellyfin Version: 10.6.4

[Working playback]
iPhone 5s, iOS 12.4.8
Jellyfin Mobile iOS 1.2.0 (5) Expo: 38.0.0

[Non-working playback]
iPhone 6s, iOS 14.0.1
Jellyfin Mobile iOS 1.2.0 (5) Expo: 38.0.0

Logs show the iPhone 6s supports Transcoding, DirectStream and DirectPlay but it goes on a loading loop and never starts, after changing tabs between "Home" and "Settings" it shows an error message saying "**This client isn't compatible with the media and the server isn't sending a compatible media format**", even though I can see transcode files being created on the server.

![playback_error](https://user-images.githubusercontent.com/6654819/94718523-38b6a080-0328-11eb-8526-ba8f76ec972f.jpeg)

**Comment by LLinoor** [comment]
The common point is that we are on iOS 14, so the problem can probably come from there.

**Comment by orhan-drsn** [comment]
I have the same issue since jellyfin server 10.6.0 on Ubuntu on all my Apple devices. 
Some movies work, some movie doesn't start and I git the playback error

**Comment by gristim** [comment]
> The common point is that we are on iOS 14, so the problem can probably come from there.

Well I'm on iOS 13 and it still doesn't work for me. However, I am able to play it through the infuse app.

**Comment by zas2** [comment]
i found out, that the issue is related to 23.976 fps movies on ios. thats the reason why some movies play and others do not. could you test and confirm?

**Comment by gristim** [comment]
I attempted to revert back to 10.5.5, but that caused some new issues. Upon a clean reinstallation, it now works. I suspect it has something to do with 10.6.x. Maybe someone can also test this out.

**Comment by ahmst** [comment]
Same problem here. I see a play icon crossed out. Only live tv works.

**Comment by mairajha** [comment]
Got this error too, what I did is to give server a time just leave it running, sometimes it will take some time to complete transcoding in background!

**Comment by linclelinkpart5** [comment]
Getting this error too, except my library is purely music, consisting of all FLAC media.
![IMG_8584](https://user-images.githubusercontent.com/844735/96488152-6b024200-11f2-11eb-8949-836e61034387.PNG)

This is what I see in the (non-debug) logs:
```[2020-10-19 09:59:21.468 -07:00] [INF] Playback stopped reported by app "Jellyfin Mobile (iOS)" "1.2.0" playing "SEQUOIA". Stopped at "0" ms
[2020-10-19 09:59:21.468 -07:00] [INF] GetPostedPlaybackInfo profile: DeviceProfile { Name: null, Id: null, Identification: null, FriendlyName: null, Manufacturer: null, ManufacturerUrl: null, ModelName: null, ModelDescription: null, ModelNumber: null, ModelUrl: null, SerialNumber: null, EnableAlbumArtInDidl: False, EnableSingleAlbumArtLimit: False, EnableSingleSubtitleLimit: False, SupportedMediaTypes: "Audio,Photo,Video", UserId: null, AlbumArtPn: null, MaxAlbumArtWidth: 0, MaxAlbumArtHeight: 0, MaxIconWidth: null, MaxIconHeight: null, MaxStreamingBitrate: 8000000, MaxStaticBitrate: 8000000, MusicStreamingTranscodingBitrate: 128000, MaxStaticMusicBitrate: null, SonyAggregationFlags: null, ProtocolInfo: null, TimelineOffsetSeconds: 0, RequiresPlainVideoItems: False, RequiresPlainFolders: False, EnableMSMediaReceiverRegistrar: False, IgnoreTranscodeByteRangeRequests: False, XmlRootAttributes: [], DirectPlayProfiles: [DirectPlayProfile { Container: "mp3", AudioCodec: "mp3", VideoCodec: null, Type: Audio }, DirectPlayProfile { Container: "aac", AudioCodec: null, VideoCodec: null, Type: Audio }, DirectPlayProfile { Container: "m4a", AudioCodec: null, VideoCodec: null, Type: Audio }, DirectPlayProfile { Container: "m4b", AudioCodec: "aac", VideoCodec: null, Type: Audio }, DirectPlayProfile { Container: "flac", AudioCodec: null, VideoCodec: null, Type: Audio }, DirectPlayProfile { Container: "alac", AudioCodec: null, VideoCodec: null, Type: Audio }, DirectPlayProfile { Container: "m4a", AudioCodec: null, VideoCodec: null, Type: Audio }, DirectPlayProfile { Container: "m4b", AudioCodec: "alac", VideoCodec: null, Type: Audio }, DirectPlayProfile { Container: "wav", AudioCodec: null, VideoCodec: null, Type: Audio }], TranscodingProfiles: [TranscodingProfile { Container: "aac", Type: Audio, VideoCodec: null, AudioCodec: "aac", Protocol: "hls", EstimateContentLength: False, EnableMpegtsM2TsMode: False, TranscodeSeekInfo: Auto, CopyTimestamps: False, Context: Streaming, EnableSubtitlesInManifest: False, MaxAudioChannels: null, MinSegments: 0, SegmentLength: 0, BreakOnNonKeyFrames: False }], ContainerProfiles: [], CodecProfiles: [], ResponseProfiles: [], SubtitleProfiles: [] }
[2020-10-19 09:59:21.495 -07:00] [INF] User policy for "mark". EnableAudioPlaybackTranscoding: True
[2020-10-19 09:59:21.495 -07:00] [INF] Bitrate exceeds DirectStream limit: media bitrate: 1021578, max bitrate: 800000
[2020-10-19 09:59:21.495 -07:00] [INF] Bitrate exceeds DirectStream limit: media bitrate: 1021578, max bitrate: 800000```

The ffmpeg logs seem correct, but I can include them as well if they could be helpful. Same with the debug logs. This is my first time trying to use Jellyfin, and not being able to play music remotely on my phone is the number 1 use case for me.

**Comment by linclelinkpart5** [comment]
Update: I was able to fix my issue of FLAC playback not working by unchecking the "Allow audio playback that requires transcoding" option for my user. Now I'm able to play music remotely on the Jellyfin iOS app.

**Comment by gnattu** [comment]
It looks like it is related to this line:
https://github.com/jellyfin/jellyfin-web/blob/9abd8ada0ef96a976ad1938977c07d0a69567c1d/src/components/appRouter.js#L475

Why the limit is that low on iOS？This is less than 1Mbps and almost can be played.

**Comment by RemoteMedia** [comment]
I have been having this same issue. Im on 10.6.4 Jellyfin Linux and I’m on iOS 14.1 and access my content remotely usually. However, this problem also occurs when on the same internal network.

I am having no issues with flac files, only mp3 files with 320kbps and 48000hz. These files work on a pc within the same internal network accessing via the same means (wifi).

I have tried in the native iOS app, and on the following browsers - brave, chrome, Firefox, safari.

I have found the issue with at least 3 files now.

The issue continues even with the amendments made to the docker container appRouter.is as suggested by @Wepwawetcode to remove the limit.

Each time I test I am saving my settings and restarting the docker container to ensure a restart has been pushed through.

I have also attempted with audio transcode on and off over all browsers with wifi and cellular to no avail.

Any other tips or confirmation would be appreciated.

**Comment by Wepwawetcode** [comment]
Which codecs are the files you are trying to play?

Verstuurd vanaf mijn iPhone

**Comment by RemoteMedia** [comment]
> Which codecs are the files you are trying to play? Verstuurd vanaf mijn iPhone
> […](#)

They are all .mp3 the flac seem to work fine. Other mp3 files at 44100hz play, it’s only the 48000hz files.

Edit:

I can actually see in the logs whether my remote access and remote transcoding is on or not that I’m having a section of the error advising

EnableRedirection=true&EnableRemoteMedia=false

I’ve also attempted and replicated this on the admin and a user account. I don’t believe this would cause issues because all my other media is playing fine.

**Comment by Wepwawetcode** [comment]
This is a strange issue and deserves a bump. I personally had an issue with all the videos being unplayable on iPhone ios, due to the bandwidth limit. For your issue I don’t have a solution yet.

Verstuurd vanaf mijn iPhone

**Comment by giacomofabbri** [comment]
> 23

I was having problem on IOS and also on Safari and that may be the problem, trying to play both mp4 and mkv with 25fps they work flawlessly! Could please someone help us find the problem? Thanks

**Comment by Freeman77** [comment]
> I was having problem on IOS and also on Safari and that may be the problem, trying to play both mp4 and mkv with 25fps they work flawlessly! Could please someone help us find the problem? Thanks

Hi, this issue should be resolved on the next release 10.7.0-rc4, in the meantime you can check out #3815 on how to make it work on 10.7.0-rc3, you have to manually replace the file Jellyfin.Api.dll

**TLDR**: If you were using a non-English OS on your server, the HLS field named FRAME-RATE could be set as "23,976" instead of "23.976", messing up playback on iOS devices. This was fixed with PR https://github.com/jellyfin/jellyfin/pull/5117
